WordPress/Classes/ViewRelated/Blog/SharingAccountViewController.swift

```diff
@@ -84,6 +112,15 @@ import WordPressShared
         var connectedAccounts = [KeyringAccount]()
         var accounts = keyringAccountsFromKeyringConnections(keyringConnections)
 
+        if accounts.count == 0 {
+            if publicizeService.externalUsersOnly && publicizeService.serviceID == PublicizeService.facebookServiceID {
```

Comment:
Would it make sense to encapsulate this boolean check in a method in `PublicizeService`, named in a way that provides a clear explanation of why we want to call `showFacebookNotice()`? 

If I read this code, without any context, I don't really know what is going on. I mean, I can understand it, but I don't really understand why this boolean check is necessary, what is the meaning of it. 

In a way, it's also leaking the abstraction a little bit. This view controller needs to know about the specific internal "state" of the `PublicizeService`, when it probably should not need to be aware of it.

Reply:
Looking a bit closer at this.  The `publicizeService.externalUsersOnly` is redundant here so we can remove it.  This will make it a check for Facebook to see if it needs special handling which will hopefully make things a bit clearer.